homeassistant/components/evohome/__init__.py

```diff
@@ -388,27 +370,105 @@ def temperature_unit(self) -> str:
         """Return the temperature unit to use in the frontend UI."""
         return TEMP_CELSIUS
 
-    async def _call_client_api(self, api_function) -> None:
+    async def _call_client_api(self, api_function, refresh=True) -> Any:
         try:
-            await api_function
+            result = await api_function
         except (aiohttp.ClientError, evohomeasync2.AuthenticationError) as err:
-            _handle_exception(err)
+            if not _handle_exception(err):
+                return
 
-        self.hass.helpers.event.async_call_later(
-            2, self._evo_broker.update()
-        )  # call update() in 2 seconds
+        if refresh is True:
+            self.hass.helpers.event.async_call_later(1, self._evo_broker.update())
 
-    async def _update_schedule(self) -> None:
-        """Get the latest state data."""
-        if (
-            not self._schedule.get("DailySchedules")
-            or parse_datetime(self.setpoints["next"]["from"]) < utcnow()
-        ):
+        return result
+
+
+class EvoChild(EvoDevice):
+    """Base for any evohome child.
+
+    This includes (up to 12) Heating Zones and (optionally) a DHW controller.
+    """
+
+    def __init__(self, evo_broker, evo_device) -> None:
+        """Initialize the evohome Controller (hub)."""
+        super().__init__(evo_broker, evo_device)
+        self._schedule = self._setpoints = {}
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._evo_device.temperatureStatus["isAvailable"]
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature of the evohome Zone."""
+        return (
+            self._evo_device.temperatureStatus["temperature"]
+            if self._evo_device.temperatureStatus["isAvailable"]
```

Comment:
This check shouldn't be necessary. The entity class will not fetch the current temperature if `available` returns `False`.

For readability we should also prefer normal if-statements over inline if-statements.

Reply:
Yeah - in a previous incarnation I'd left `self.available = True` at all times.  The issue is that all tracking of state stops for the device when self.available = False...

I think I will revert to `self.available = True`...